var/spack/repos/builtin/packages/metall/package.py

```diff
@@ -29,9 +30,12 @@ class Metall(CMakePackage):
 
     depends_on('cmake@3.10:', type='build')
     depends_on('boost@1.64:', type=('build', 'link'))
+
     # googletest is required only for test
-    # 'make test' is executed if '--run-tests' or '--test=root' is specified
+    # GCC is also required only for test (Metall is a header-only library)
+    # Memo: type 'spack install --test=root metall' or 'spack install --test=all metall' to run test ('make test' is launched)
     depends_on('googletest', type=('test'))
+    depends_on('gcc@8.1.0:', type=('test'))
```

Comment:
On a Ubuntu 18.04 with just the default gcc-7.5.0 installed, you'd still get this:
spack install -v --test=root metall 
```py
  >> 18    CMake Error at CMakeLists.txt:209 (message):
     19      GCC version must be at least 8.1
```
The reason is that this does not change the used compiler. It provides gcc@8.1.0 like it would provide a library, but not set up as a compiler. To use a recipe for as a compiler for compiling the code, spack uses the `%` as prefix, so `%gcc@8.1.0:` would be a compiler spec.

But it results in a syntax error to use it alone. You can apply it to another dependency, like googetest in metall, and of course it is used for the entire build, not just the single dependency (and in this case only for `--test`)

This is proposal which you can apply by clicking the button labelled **"commit suggestion"** below does what you want.
```suggestion
    # Hint: Use 'spack install --test=root metall' or 'spack install --test=all metall'
    # to run test (adds a call to 'make test' to the build)
    depends_on('googletest %gcc@8.1.0:', type=('test'))
```
You also have to wrap the long line at column 88, which this suggestion does as well.

Reply:
Thank you for your feedback!